### ISSUE TITLE
CABLE as used in ESM16 build

### ISSUE THREAD

**Issue by JhanSrbinovsky** [CABLE-LSM/CABLE#466]
This is not intended to merge into main (yet). Corresponds to require modifications to CABLE:main files in ESM1.5+CABLE3.

cable_surface_type modifications are also coming through another PR (#446, #467) directly to CABLE:main.

https://github.com/ACCESS-NRI/ACCESS-ESM1.6/pull/5#issue-2606993820

**Comment by JhanSrbinovsky** [comment]
Screenshot 2024-11-08 at 2.27.27 pm

**Comment by JhanSrbinovsky** [comment]
File added:
src/coupled/ESM1.5/cable_surface_types.F90
AT 1f7fff9  - added same as version in offline/

Files modified:

  src/coupled/ESM1.5/CABLEfilesFromESM1.5/cable_cbm.F90
  src/coupled/ESM1.5/CABLEfilesFromESM1.5/cable_rad_driver.F90
  src/params/grid_constants_cbl.F90
  src/science/canopy/cbl_SurfaceWetness.F90
  src/science/canopy/cbl_init_wetfac_mod.f90
  src/science/casa-cnp/bgcdriver.F90
  src/science/casa-cnp/casa_inout.F90
  src/science/casa-cnp/casa_rplant.F90
  src/science/misc/cable_climate.F90
  src/science/roughness/cable_roughness.F90
  src/science/soilsnow/cbl_surfbv.F90

minor mods. mostly adjust 4 #FPP directive UM_CBL declaration = true AND cable_surface_types SHOULD exist by now

**Comment by ccarouge** [comment]
Why are you modifying files under the ESM1.5 directory? I would believe these should reflect what is needed in ESM1.5 and not ESM1.6. Additionally, I believe @bschroeter and you are doing the same work right now.

**Comment by JhanSrbinovsky** [comment]
hrmmmm - not sure what you mean. we are already running ESM1.6. And it all started from ESM1.5+CABLE3 and we just haven't renamed the directory. I'm just  noting the changes relative to this main brach of CABLE. What is Ben doing, I thought he was working on the library thing.

**Comment by JhanSrbinovsky** [comment]
You're right that we would need to establish a 2nd ESM1.6 directory if indeed these excluded ESM1.5 from running and/or we wanted the ESM1.5 directory to reflect the CMIP6 model. The former, I can't vouch for because I haven't gone backwards and tried to use this ESM1.5/ layer of files with CABLE2.4 (although I think there would be a reasonable chance of them working). On the latter, they are not far removed from the CMIP6 versions, although not exactly the same. Further muddying the waters, apart from very few people having modified the ESM1.5 code, I suspect that no-one other than myself has actually started from the code in src/coupled/ESM1.5/ etc. I mean they almost certainly use the ESM1.5 code that already contains all those things. As in the UM7 repo as well. (BTW - this here corresponds to my branch in the UM7 repo, which in turn is picked up in the ESM1.6 build. I'm just discussing now with Aidan why my PAYU cloning is not working.)  Perhaps then we should just dismiss the ESM1.5 directory and call it in ESM1.6. Same with ACCESS-CM2, although the is even messier as there is a JULES repo hauled into the build and CABLE is a part of that

**Comment by JhanSrbinovsky** [comment]
This builds through our old script based method which is where it was developed. I was also pursuing ESM16 builds at the time. I got this to build through ESM16 repo. The corresponding run through payu I couldn't get to work.  I had previously run it using our old script based method, and I had some testing to do on another front so I returned to this method to get the testing done. It took me all day just to get the original to run. This modified main branch is still not running and the logs are next to useless. Trying debug build etc.

**Comment by JhanSrbinovsky** [comment]
UM7 repo is very close to spinup stage
